### DISCUSSION TITLE
What are the conditions for a module to be in Core?

### DISCUSSION THREAD

**Discussion by Jarod42** [premake/premake-core#1689] [General]
What are the conditions for a module to be in Core?

You removed CodeBlocks, keep Codelite, gmake, msvc.

I wonder the condition to be in Core (or embdedded in Release).

**Comment by starkos** [comment]
There's been a bit of tug of war about that; it is a judgement call. The big deciding factor *for me* is whether or not we have the resources to support and maintain it properly. Based on what I've seen so far, we're better off keeping the number of core modules small and improving support and discoverability for third-party modules instead.

I think Visual Studio, Xcode, and makefiles have to be in core. D support should IMHO not be there. The others are nice to have when folks are around to help support them, but really problematic when they aren't.

That doesn't really answer your question I'm afraid. With the 6.x work I'm trying to really think through all of the APIs to give module developers a stable platform. Once we've ported the big three exporters above I'd love to improve the module support. And we've seen a slow but steady influx of new contributors like yourself—if that continues it will be much easier to allow new modules into core.

Hope that helps a little?

**Comment by Jarod42** [comment]
Currently, it is hard to know what works with modules (not only external ones BTW). And unfortunately, all external modules I tested have very minimal support, even for basic features :-/

I'm not sure how 6.x might change that.

As user, we might expect that the core one are well supported, and I understand than adding one in Core requires resource you don't have :-/.

I wonder if git-submodule might allow to still keep Core minimal (or as-is) but allows to release with more modules embedded.

Thanks for answer.

**Comment by starkos** [comment]
> I wonder if git-submodule might allow to still keep Core minimal (or as-is) but allows to release with more modules embedded.

I forced that approach on the community when we first moved to GitHub—it didn't go well and we ended up folding things back into a single repository. You can still see the original, archived submodule repositories under the Premake team account. It didn't solve any of the problems with modules and made the whole development process more troublesome.

> Currently, it is hard to know what works with modules (not only external ones BTW)

I'm intending to do a better and more thorough job of documenting what's supported where as I build out 6.x, at least for the core modules. Feel free to call me on it if you see gaps.
